### PR TITLE
[REVIEW] Add cudf.DataFrame.mean = None

### PR DESCRIPTION
**Pull request by mrocklin**
This resolves a version dispute between dask.dataframe, cudf, and
dask-cudf

### CODE REVIEW

**Comment by kkraus14** [comment]
cc @mike-wendt @raydouglass this is a hotfix PR

**Comment by raydouglass** [comment]
@mrocklin Can you check out the unit test failure?

**Comment by mrocklin** [comment]
I wouldn't worry about it.  It only affects testing code and is fixed upstream in Dask.

**Comment by kkraus14** [comment]
> I wouldn't worry about it. It only affects testing code and is fixed upstream in Dask.

If that's the case why aren't we pinning to a version of Dask that resolves the issue?

**Comment by mrocklin** [comment]
You could, but there isn't really an issue here except in testing.  Generally I try to keep fairly open-ended requirements when publishing infrastructural libraries.

Alternatively, if this is very important, then you could also wait a couple days for the next Dask release.

**Comment by mrocklin** [comment]
I don't care though.  Please feel free to do whatever you think is best here.

**Comment by kkraus14** [comment]
> I don't care though. Please feel free to do whatever you think is best here.

If the tests are broken and will be broken if a user uses any version of dask except the newest that sounds like an issue to me. We should either pin the version(s) of dask to something that passes all of our tests, or we should fix the tests to work with the versions of dask we support. Otherwise a user could take a similar codepath to what exists in the test(s) and have a poor experience.

**Comment by mrocklin** [comment]
> user uses any version of dask except the newest that sounds like an issue to me.

It's only an issue if they use the same `assert_eq` function that we use for testing.  I don't think that users commonly do this.  They do, however, more often ask for newer versions of things.  We're going to fix some other cudf/dask bug in the next few weeks, and if we add a version constraint now they those users won't be able to get that fix.  It's not perfect either way.

> Otherwise a user could take a similar codepath to what exists in the test(s) and have a poor experience.

I don't think that this is True.  There was an explicit `isinstance(..., pd.Index)` call in our testing code that needed to be changed to an `is_index_like(...)` call.  The change wasn't in Dask or cudf source code.  It was in the testing infrastructure.

**Comment by mrocklin** [comment]
But, if you want me to set a different version number somewhere, I'm happy to make that change.  Alternatively, if you want to make that change I'd also be very happy to have you do that.

**Comment by kkraus14** [comment]
@mrocklin based on the reasoning, would you be opposed to me adding a small bit of logic to this PR that checks the Dask version and skips the failing pytest(s) if the version is less than a certain number? From my perspective that's a minimal change that gets the best of all worlds.

**Comment by mrocklin** [comment]
Go for it.   Thanks for the help.

File: dask_cudf/tests/test_core.py [hunk 1]
```diff
@@ -327,6 +327,14 @@ def gddf(gdf):
 def test_unary_ops(func, gdf, gddf):
     p = func(gdf)
     g = func(gddf)
+
+    # Fixed in https://github.com/dask/dask/pull/4657
+    if isinstance(p, cudf.Index):
+        from packaging import version
+        if version.parse(dask.__version__) < version.parse("1.1.6"):
```
Comment (kkraus14) [inline dask_cudf/tests/test_core.py]: @mrocklin can you confirm this is the correct version with the fix?

Comment (mrocklin) [reply dask_cudf/tests/test_core.py]: Sure.  That should work.

**Pull request merged**